bin/proving-service/build.rs

```diff
@@ -1,93 +1,42 @@
-/// environment variable.
+/// Generates Rust protobuf bindings.
 fn main() -> miette::Result<()> {
     println!("cargo::rerun-if-env-changed=BUILD_PROTO");
     if !BUILD_GENERATED_FILES_IN_SRC {
         return Ok(());
     }
 
-    copy_proto_files()?;
-    compile_tonic_server_proto()
-}
-
-// HELPER FUNCTIONS
-// ================================================================================================
+    // Get both file descriptor sets
+    let worker_status_descriptor = worker_status_api_descriptor();
+    let proving_service_descriptor = proving_service_api_descriptor();
 
-/// Copies the proto file from the root proto directory to the proto directory of this crate.
-fn copy_proto_files() -> miette::Result<()> {
-    // remove and create dirs
-    fs::remove_dir_all(CRATE_PROTO_DIR).into_diagnostic()?;
-    fs::create_dir_all(CRATE_PROTO_DIR).into_diagnostic()?;
-
-    for file in PROTO_FILES {
-        let src = format!("{REPO_PROTO_DIR}/{file}");
-        let dest = format!("{CRATE_PROTO_DIR}/{file}");
-        fs::copy(src, dest).into_diagnostic()?;
-    }
+    // Build tonic code for each descriptor
+    build_tonic_from_descriptor(worker_status_descriptor)?;
+    build_tonic_from_descriptor(proving_service_descriptor)?;
 
     Ok(())
 }
 
-fn compile_tonic_server_proto() -> miette::Result<()> {
-    let crate_root =
-        PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should be set"));
-    let dst_dir = crate_root.join("src").join("generated");
-
-    // Remove generated files if they exist
-    for file in PROTO_FILES {
-        let _ = fs::remove_file(dst_dir.join(file.replace(".proto", ".rs")));
-    }
-
-    let out_dir = env::var("OUT_DIR").into_diagnostic()?;
-    let file_descriptor_path = PathBuf::from(out_dir).join("file_descriptor_set.bin");
-
-    let proto_dir: PathBuf = CRATE_PROTO_DIR.into();
-    let protos: Vec<PathBuf> = PROTO_FILES.iter().map(|file| proto_dir.join(file)).collect();
-    let includes = &[proto_dir];
-
-    let file_descriptors = protox::compile(&protos, includes)?;
-    fs::write(&file_descriptor_path, file_descriptors.encode_to_vec()).into_diagnostic()?;
-
-    build_tonic_server(&file_descriptor_path, &dst_dir, &protos, includes)?;
-
-    Ok(())
-}
+// HELPER FUNCTIONS
+// ================================================================================================
 
-fn build_tonic_server(
-    file_descriptor_path: &Path,
-    out_dir: &Path,
-    protos: &[PathBuf],
-    includes: &[PathBuf],
-) -> miette::Result<()> {
+/// Builds tonic code from a `FileDescriptorSet`
+fn build_tonic_from_descriptor(descriptor: FileDescriptorSet) -> miette::Result<()> {
     tonic_build::configure()
-        .file_descriptor_set_path(file_descriptor_path)
-        .skip_protoc_run()
-        .out_dir(out_dir)
-        .build_server(true)
+        .out_dir(GENERATED_OUT_DIR)
+        .build_server(true) // this setting generates only the client side of the rpc api
```

Comment:
Is the comment here accurate? It seems like this should generate only the server code - or am I missing something?